src/dmd/parse.d

```diff
@@ -9564,3 +9302,276 @@ enum PREC : int
     unary,
     primary,
 }
+
+/**********************************
+ * Set operator precedence for each operator.
+ *
+ * Used by hdrgen
+ */
+immutable PREC[EXP.max + 1] precedence =
```

Comment:
hdrgen.d is accessing this quite a bit so it's not private.

Something is smelling here I think, although I'm not quite sure what.

Reply:
I'm not really sure where `precedence` should go. I moved it to the end of the public section because I almost never want to look at it. Also, moving code refactors should not try "fixing" things.

Reply:
Yeah, it needed to be commented on but this isn't a review so it shouldn't hold up this PR.

From a very cursory glance, it looks like from an architecture POV like an output stage of a utility program depending on an input stage. What this makes me think of is that this should be in a common library dependency rather than in either of them.

Apart from cli/globals stuff, does dmd-fe have anything else like that?

Reply:
I think that `precedence` should go in `tokens.d`. `tokens.d` is imported by both the parser and the header generator and, in my opinion, it logically makes sense to it there.